### PR TITLE
CSHARP-4832: Support sort by score $search stage

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs [hunk 1]
```diff
@@ -447,6 +447,22 @@ public void Sort()
...
+                .Search(
+                    Builders<Movie>.Search.QueryString(x => x.Title, "dance"),
+                    new() { Sort = Builders<Movie>.Sort.MetaSearchScoreAscending() })
+                .Project<Movie>(Builders<Movie>.Projection
+                    .Include(x => x.Title)
+                    .MetaSearchScore("score"))
+                .Limit(10)
+                .ToList();
+            results.First().Title.Should().Be("Invitation to the Dance");
+            results.Should().BeInAscendingOrder(m => m.Score);
```
Comment (adelinowona) [inline tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:463]: Should we also test MetaSearchScoreDescending? Or is MetaSearchScoreAscending enough?

**Review by adelinowona — APPROVED** [review]
LGTM

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs (hunk 1 above)
Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:463]: `AtlasSearchTests` is scoped to contain simple tests of each operator and some of option. 
`MetaSearchScoreAscending` and `MetaSearchScoreDescending` translate to same MQL option, therefore it's sufficient to test the different variations of that in unitests.

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**